### PR TITLE
Feat custom email icons

### PR DESCRIPTION
**Pull request by HenryRoutson**
## What kind of change does this PR introduce?

Allowing no or custom prefix icons / widgets

## What is the current behavior?

Prefix icons are set

## What is the new behavior?

Prefix icons can be removed or changed

## Additional context

I would have put an issue, but this was almost quicker.
The reason for the pull request is simply that the design i want to create doesn't have icons.

I also made a small simplification with the navigateHome function

Thanks!

### CODE REVIEW

**Comment by HenryRoutson** [comment]
This is the sort of design i'm looking to create

login-design-examples

**Comment by HenryRoutson** [comment]
I've added a minimal dark mode example if you want,
it more closely matches the supabase react auth ui
https://docs-git-docs-add-next-js-server-component-docs-supabase.vercel.app/docs/guides/auth/auth-helpers/auth-ui

Screenshot 2024-08-11 at 9 31 32 pm

File: lib/src/components/supa_email_auth.dart [hunk 1]
```diff
@@ -95,6 +95,10 @@ class SupaEmailAuth extends StatefulWidget {
   /// Localization for the form
   final SupaEmailAuthLocalization localization;
 
+  /// Icons or custom prefix widgets for email UI
+  final Widget? prefixIconsEmail;
+  final Widget? prefixIconsPassword;
```
Comment (dshukertjr) [inline lib/src/components/supa_email_auth.dart:100]: Would it make sense to make it singular?
```suggestion
  final Widget? prefixIconEmail;
  final Widget? prefixIconPassword;
```

**Review by dshukertjr — COMMENTED** [review]
Thanks for such an amazing PR! I love it! It looks very nice!

File: example/lib/sign_in.dart [hunk 2]
```diff
@@ -6,22 +6,48 @@ import 'constants.dart';
...
 class SignUp extends StatelessWidget {
   const SignUp({Key? key}) : super(key: key);
-
   @override
   Widget build(BuildContext context) {
+    void navigateHome(AuthResponse response) {
+      Navigator.of(context).pushReplacementNamed('/home');
+    }
+
+    var darkModeThemeData = ThemeData.dark().copyWith(
```
Comment (dshukertjr) [inline example/lib/sign_in.dart:15]:
```suggestion
    final darkModeThemeData = ThemeData.dark().copyWith(
```

**Comment by dshukertjr** [comment]
Would you be able to update the variable names on the places that they are referenced?

**Comment by HenryRoutson** [comment]
Yep, done

**Review by dshukertjr — APPROVED** [review]
Thanks for this PR!

**Pull request merged**